subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformer.java

```diff
@@ -72,8 +74,14 @@ public DefaultTransformer(Class<? extends ArtifactTransform> implementationClass
         return outputs;
     }
 
-    private ArtifactTransform newTransformer() {
-        return instantiator.newInstance(implementationClass, parameters.isolate());
+    private ArtifactTransform newTransformer(ArtifactTransformDependencies artifactTransformDependencies) {
+        DefaultServiceRegistry registry = new DefaultServiceRegistry();
```

Comment:
If creating the service registry is expensive, should we create the registry only when the dependencies are actually required by the transformer?

Reply:
I'd change things around and make a cheap service registry instead. Let's address that in a separate PR.